store/tikv/region_cache.go

```diff
@@ -392,11 +426,15 @@ func (c *RegionCache) GetTiKVRPCContext(bo *Backoffer, id RegionVerID, replicaRe
 		storeIdx  int
 		accessIdx AccessIndex
 	)
+	options := &storeSelectorOp{}
+	for _, op := range opts {
+		op(options)
+	}
 	switch replicaRead {
 	case kv.ReplicaReadFollower:
-		store, peer, accessIdx, storeIdx = cachedRegion.FollowerStorePeer(regionStore, followerStoreSeed)
+		store, peer, accessIdx, storeIdx = cachedRegion.FollowerStorePeer(regionStore, followerStoreSeed, options)
 	case kv.ReplicaReadMixed:
-		store, peer, accessIdx, storeIdx = cachedRegion.AnyStorePeer(regionStore, followerStoreSeed)
+		store, peer, accessIdx, storeIdx = cachedRegion.AnyStorePeer(regionStore, followerStoreSeed, options)
```

Comment:
I don't think stale read should be compatible with follower read in this way.
Users always want to read the nearest replica in stale read, regardless of its Raft role.
But to this end, users need to either enable `ReplicaReadFollower` or `ReplicaReadMixed`.

Reply:
I agree with the opinion that `stale read don't have to be compatible with follower read`. I just think `GetTiKVRPCContext` don't  need to care about it. `GetTiKVRPCContext ` just give the ability to select the peers in many diamensions and each diamension is orthogonal to each other.

When `stale read` call the `GetTiKVRPCContext`, it could call it with `MixedRole` and label selectors together without user pre-enabling.